src/Common/dot_wrapper.h

```diff
@@ -0,0 +1,29 @@
+#ifndef DOT_WRAPPER_H
+#define DOT_WRAPPER_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifdef _WIN32
+#define inline __inline
+#endif
+
+#ifdef CARET_DOTFCN
+#include "dot.h"
```

Comment:
In the case of the command line tools, the caller could compare the return value to the input when calling dot_set_impl() and issue a warning as appropriate. In the case of the GUI, the cpuinfo functions could be used to find out about the supported instruction set extensions before even calling any of the dot module functions so that the unsupported options could be grayed out in the corresponding GUI control. Another way to find out which variants are supported (without using cpuinfo directly) would be to call dot_set_impl(DOT_AVX) and look at the return value. All variants with a value (in the enum) <= that return value are supported.

I have updated my workbench fork's dot module now.